src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py

```diff
@@ -641,7 +650,8 @@ def flexible_server_provision_network_resource(cmd, resource_group_name, server_
         raise RequiredArgumentMissingError("Private DNS zone can only be used with private access setting. Use vnet or/and subnet parameters.")
     else:
         start_ip, end_ip = prepare_public_network(public_access, yes=yes)
-
+        if public_access is not None and str(public_access).lower() == 'Disabled'.lower():
```

Comment:
I am little bit confused for these two lines of code. Is network.public_network_access case-sensitive or not? Why line 566-567 are made by 
if public_access.lower() == "disabled" or public_access.lower() == "enabled":
            instance.network.public_network_access = public_access
But these two lines are different, while they are in same purpose.